lib/html/static/js/fileSearch.js

```diff
@@ -0,0 +1,143 @@
+		for (i=0; i<kwargs.length; i++) {
+			// check for js null and python None
+			if (kwargs[i][1] && kwargs[i][1] != 'None') {
+				var c = '&';
+				if (is_first) {
+					c = '?';
+					is_first = false;
+				}
+				url += c + kwargs[i][0] + '=' + kwargs[i][1];
+			}
+		}
+		return url;
+	}
+
+
+    var loadFile = function(search_string, goto) {
+		/* Loads in a file from the provided url, populates the line
+		 * numbers and scrolls to the goto line is provided. */
+		
+		// update previous search
+		previous_search = search_string;
+		// detect search mode
+		search_mode = $("input:radio[name=search_mode]:checked").val();
+		// generate the search url
+		var base = '/file_view/' + user + '/' + suite;
+		var args = [['path', path], ['path_in_tar', path_in_tar],
+			['search_string', search_string], ['search_mode', search_mode]];
+		var url = buildUrl(base, args);
+
+		// asynchronously load file
+        $.ajax({
+            url: url,
+            success: function(result) {
+                $('#file-container').html(result);
+                lineNumbers = false;
+                if (goto) {
+	            selectLine(goto);
+		}
+            },
+            error: function(error) {
+                alert('Search failed!\nerror opening: ' + url);
+            }
+        });
+    }
+
+
+    var selectLine = function(num) {
+        /* Highlights and scrolls to the line of code with the provided
+		 * number. */
+        // insert highlight tags to selected line
+        lines = $('#filecode').html().split('\n');
+        if (num<lines.length) {
+            lines[num-1] = mark[0] + lines[num-1] + mark[1];
+        }
+        $('#filecode').html(lines.join('\n'));
+        // scroll to selected line
+        $('html, body').animate({scrollTop: $('#'+num).offset().top}, 500);
+    }
+
+
+    this.gotoLine = function(num) {
+        /* Reloads the whole file and propts loadFile() to scroll to the
+         * specified line. */
+		loadFile("", num);
+		previous_search = null;
+    }
+
+
+    this.fileSearch = function() {
+        /* Called on submit by form:file_search to initiate a search. */
+        var search_string = document.file_search.search_string.value;
+		if (search_string == previous_search)
+			return false;
+        if (search_string != "" && search_string != null) {
+	    	loadFile(search_string);
+        }
+        else {
+            loadFile("");
+        }
+        return false;
+    }
+
+
+	this.init = function() {
+		var form = document.forms['file_search'];
+		user = form['user'].value;
+		suite = form['suite'].value;
+		path = form['path'].value;
+		path_in_tar = form['path_in_tar'].value;
+	}
+
+
+    var self = {
+        /* Export public functions. */
+        gotoLine: this.gotoLine,
+        fileSearch: this.fileSearch,
+		init: this.init
+    }
+
+    return self;
+
+})();
+
+
+// load file once all resources (incl FileLoader) have been obtained
+$(function() {
+	FileLoader.init();
+	FileLoader.fileSearch();
+});
```

Comment:
You should pass this file through something like JSLint. There is a number of indentation and spacing issues in this file.